frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RenameBackgroundable.java

```diff
@@ -78,6 +84,57 @@ public void run(ProgressIndicator indicator) {
     // (Stepping deeper is not an option since we would lose some important logic or become very dependent on the internals of git4idea).
     // Hence, we wait for the creation of the branch (with exponential backoff).
     waitForCreationOfLocalBranch(gitRepository, newBranchName);
+
+    if (gitNewBranchOptions.shouldSetTracking()) {
+      val remote = currentBranchSnapshot.getRemoteTrackingBranch();
+      assert remote != null : "shouldSetTracking is true but the remote branch does not exist";
+      handleSetTrackingBranch(remote.getRemoteName());
+    }
+  }
+
+  @UIThreadUnsafe
+  private void handleSetTrackingBranch(String remoteName) {
+    val newBranchName = gitNewBranchOptions.getName();
+    val renamedRemoteName = "${remoteName}/${newBranchName}";
+
+    // setUpstream: git branch --set-upstream-to <upstreamBranchName> <branchName>
+    val setUpstream = Git.getInstance().setUpstream(gitRepository, renamedRemoteName, newBranchName);
+    var error = setUpstream.getErrorOutputAsJoinedString();
+
+    if (!setUpstream.success()) {
+      if (setUpstream.getErrorOutput().get(0)
+          .equals("fatal: the requested upstream branch '${renamedRemoteName}' does not exist")) {
+        val unsetUpstream = unsetUpstream(newBranchName);
+        if (unsetUpstream.success()) {
+          // This happens when one renames a branch with a remote branch set up,
+          // and a local branch with the new name is created for the second and next time,
+          // and the remote with the new name does not exist.
```

Comment:
Huh could you provide a sequence of `git` commands in this comment  that would lead to such an effect? that's pretty hard to follow 🤔

Reply:
IDE git console shows `git branch -m ...` calls only.

You can simply reproduce it.
1. `./scripts/setup-multiroot-sandbox`
2. `runIde` 
3. Rename `allow-ownership-link` to `allow-ownership-link-XD` with the option to rename remote
4. Rename `allow-ownership-link-XD` to `allow-ownership-link` with the option to rename remote 
5. Rename `allow-ownership-link` to `allow-ownership-link-XD` with the option to rename remote